src/context.rs

```diff
@@ -0,0 +1,87 @@
+use http::{Extensions, HeaderMap, Method, Uri, Version};
+use http_service::Body;
+use route_recognizer::Params;
+use std::sync::{Arc, Mutex};
+
+#[derive(Clone)]
+pub struct Context<AppData> {
+    app_data: Arc<AppData>,
+    context_data: Arc<ContextData>,
+}
+
+struct ContextData {
+    meta: http::request::Parts,
+    body: Mutex<http_service::Body>,
+    extensions: Mutex<Extensions>,
```

Comment:
I think `Context` isn't required to be `Send` nor `Sync` because only one thread will be accessing the struct during the lifetime (as each request is processed in its own thread.) Did I get it wrong?

Reply:
I would not agree. It's certainly rare, but also not unusual to spawn further threads during a request execution that might have access to the request data.